src/renderer/components/wallet/account/AccountSelector.tsx

```diff
@@ -28,19 +28,31 @@ const filterFunction = ({ asset }: WalletBalance, searchTerm: string) => {
 }
 
 export const AccountSelector: React.FC<Props> = (props): JSX.Element => {
-  const { selectedAsset, walletBalances, onChange = (_) => {}, size = 'large', network } = props
+  const { selectedWallet, walletBalances, onChange = (_) => {}, size = 'large', network } = props
 
   const intl = useIntl()
 
   const filteredWalletBalances = useMemo(
     () =>
       walletBalances.filter(
-        ({ asset, amount }) => asset.symbol !== selectedAsset.symbol && amount.amount().isGreaterThan(0)
+        ({ asset, amount }) => asset.symbol !== selectedWallet.asset.symbol && amount.amount().isGreaterThan(0)
       ),
-    [walletBalances, selectedAsset]
+    [selectedWallet.asset.symbol, walletBalances]
   )
   const enableDropdown = filteredWalletBalances.length > 0
 
+  const isLedgerWalletType = useMemo(
+    () =>
+      walletBalances.filter(
+        ({ asset, amount, walletType, walletAddress }) =>
+          walletAddress === selectedWallet.walletAddress &&
+          asset.symbol === selectedWallet.asset.symbol &&
+          amount.amount().isGreaterThan(0) &&
+          walletType === 'ledger'
+      ).length === 1,
+    [walletBalances, selectedWallet.walletAddress, selectedWallet.asset.symbol]
+  )
```

Comment:
Why do we need this ^ if we have `walletType` in `selectedWallet`?

Reply:
Good catch! 👍
Before the last commit we passed `selectedAsset`, not `selectedWallet` so it was needed.